alicat/serial.py

```diff
@@ -280,11 +422,98 @@ def set_pressure(self, pressure, retries=2):
                 purchase. Likely in psia.
         """
-        if self.control_point is not None and self.control_point != 'pressure':
+        if any([self.control_point is not None and
+                self.control_point == 'mass flow',
+                self.control_point is not None and
+                self.control_point == 'vol flow']):
             self._set_setpoint(0, retries)
-            self._set_control_point('pressure', retries)
+            self._set_control_point('abs pressure', retries)
         self._set_setpoint(pressure, retries)
 
+    def hold(self, retries=2):
+        """Override command to issue a valve hold.
+
+        For a single valve controller, hold the valve at the present value.
+        For a dual valve flow controller, hold the valve at the present value.
+        For a dual valve pressure controller, close both valves.
+        """
+        self._test_controller_open()
+        command = '{addr}$$H\r'.format(addr=self.address)
+        self._write_and_read(command, retries)
+
+    def cancel_hold(self, retries=2):
+        """Cancel valve hold."""
+        self._test_controller_open()
+        command = '{addr}$$C\r'.format(addr=self.address)
+        self._write_and_read(command, retries)
+
+    def read_PID(self, retries=2):
+        """Read the current PID values on the controller.
+
+        Values include the loop type, P value, D value, and I value.
+        Values returned as a dictionary.
+        """
+        self._test_controller_open()
+
+        self.pid_keys = ['loop_type', 'P', 'D', 'I']
+
+        command = '{addr}$$r85\r'.format(addr=self.address)
+        read_loop_type = self._write_and_read(command, retries)
+        spl = read_loop_type.split()
+
+        loopnum = int(spl[3])
+        loop_type = ['PD/PDF', 'PD/PDF', 'PD2I'][loopnum]
+        pid_values = [loop_type]
+        for register in range(21, 24):
+            value = self._write_and_read('{}$$r{}\r'.format(
+                self.address,
+                register))
+            value_spl = value.split()
+            pid_values.append(value_spl[3])
+
+        result = {k: (v if k == self.pid_keys[-1] else str(v))
+                  for k, v in zip(self.pid_keys, pid_values)}
+
+        return result
+
+    def write_PID_looptype(self, looptype, retries=2):
+        """Change the PID loop from PD/PDF to PD2I and vice versa.
+
+        Done by changing the appropriate bits in register 85.
+        """
+        self._test_controller_open()
+
+        command = '{addr}$$w85={loop_num}\r'.format(
+            addr=self.address,
+            loop_num=['', 'PD/PDF', 'PD2I'].index(looptype)
+        )
+
+        self._write_and_read(command, retries)
+
+    def write_PID_P(self, p_value, retries=2):
+        """Change P value for PID tuning."""
+        self._test_controller_open()
+        value = p_value
+        command = '{addr}$$w21={v}\r'.format(addr=self.address, v=value)
```

Comment:
You could save a line on this and just do:

```python
command = '{addr}$$w21={v}\r'.format(addr=self.address, v=p_value)
```

Also applies to below functions.